File: support/mock_backend/devLib/ds1302.c
#include <stdint.h>
#include <stddef.h>
#include <stdbool.h>
#include <string.h>
#include "devLib/ds1302.h"
unsigned int ds1302rtcRead(const int reg) { (void)reg;return 1; }
void ds1302rtcWrite(const int reg, const unsigned int data) { (void)reg;(void)data; }
unsigned int ds1302ramRead(const int addr) { (void)addr;return 1; }
void ds1302ramWrite(const int addr, const unsigned int data) { (void)addr;(void)data; }
void ds1302clockRead(int * clockData) { for(int i=0;i<8;++i)clockData[i]=10+i; }
void ds1302clockWrite(const int * clockData) { (void)clockData; }
void ds1302trickleCharge(const int diodes, const int resistors) { (void)diodes;(void)resistors; }
void ds1302setup(const int clockPin, const int dataPin, const int csPin) { (void)clockPin;(void)dataPin;(void)csPin; }

File: tests/devLib/ds1302.py
import importlib
import pytest

m=importlib.import_module("wiringop.dev_lib.ds1302")

def test_ds1302rtc_read():
    _=m.ds1302rtc_read(1)

def test_ds1302rtc_write():
    _=m.ds1302rtc_write(1, 1)

def test_ds1302ram_read():
    _=m.ds1302ram_read(1)

def test_ds1302ram_write():
    _=m.ds1302ram_write(1, 1)

def test_ds1302_clock_read():
    assert m.ds1302_clock_read()==tuple(range(10,18))

def test_ds1302_clock_write():
    m.ds1302_clock_write(range(8))
    with pytest.raises(ValueError):m.ds1302_clock_write([1])

def test_ds1302trickle_charge():
    _=m.ds1302trickle_charge(1, 1)

def test_ds1302setup():
    _=m.ds1302setup(1, 1, 1)
